### PR TITLE
Fix type declarations for custom message arguments

### LINKED ISSUE #6353: Tyep declarations shipped in 14.12.0 are invalid
### What steps are needed to reproduce the bug?

Write a custom rule importing stylelint
Run the typechecker on that file (which will import the stylelint types).
This will report an error in the stylelint type

### What Stylelint configuration is needed to reproduce the bug?

n/a

### How did you run Stylelint?

n/a

### Which version of Stylelint are you using?

14.12.0

### What did you expect to happen?

No type checking error

### What actually happened?

```
node_modules/stylelint/types/stylelint/index.d.ts:356:29 - error TS2344: Type 'RuleMessage' does not satisfy the constraint '(...args: any) => any'.
  Type 'string' is not assignable to type '(...args: any) => any'.

356    messageArgs?: Parameters<RuleMessage> | undefined;
                                ~~~~~~~~~~~


Found 1 error in node_modules/stylelint/types/stylelint/index.d.ts:356
```

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

Use `Parameters<RuleMessageFunc>` instead of `Parameters<RuleMessage>` (`RuleMessage` is `string | RuleMessageFunc`) and fix the type linting to actually report such errors.

### PR DESCRIPTION
**Pull request by stof**
> Which issue, if any, is this issue related to?

Closes #6353

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: .changeset/eleven-monkeys-return.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fix type declarations for custom message arguments
```
Comment (ybiquitous) [inline .changeset/eleven-monkeys-return.md:5]: [suggest] Thanks for adding the entry. Please fit the prefix to the existing format in `CHANGELOG.md` as below:

```suggestion
Fixed: type declarations for custom message arguments
```

File: tsconfig.json [hunk 2]
```diff
@@ -15,7 +15,6 @@
 		"noUncheckedIndexedAccess": true,
 		"esModuleInterop": true,
 		"resolveJsonModule": true,
-		"skipLibCheck": true,
```
Comment (ybiquitous) [inline tsconfig.json:18]: [suggest] Can you change `true` to `false` to make it easier for us to find a commit history?

```diff
-"skipLibCheck": true,
+"skipLibCheck": false,
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@stof Thanks for your first PR contribution. I couldn't notice the type error due to `skipLibCheck: false`.

I appreciate your report right after the 14.12.0 release. 👍🏼

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Comment by stof** [comment]
@ybiquitous do you have any estimate about when the fix will be released ?

**Comment by ybiquitous** [comment]
@stof Perhaps within a few days.

**Comment by stof** [comment]
thanks

**Comment by ybiquitous** [comment]
[14.12.1](https://github.com/stylelint/stylelint/releases/tag/14.12.1) is released!

**Pull request merged**